response_operations_ui/contexts/reporting_units.py

```diff
@@ -0,0 +1,231 @@
+        collection_exercises, ru["sampleUnitRef"], survey["shortName"], permissions["reporting_unit_edit"]
+    )
+    respondents_section = _build_respondents_section(
+        respondents,
+        case["id"],
+        collection_exercises[0]["id"],
+        collection_exercises[0]["tradingAs"],
+        ru["name"],
+        ru["sampleUnitRef"],
+        ru["id"],
+        survey["surveyRef"],
+        survey["shortName"],
+        survey["id"],
+        unused_iac,
+        permissions,
+    )
+
+    return {"collection_exercise_section": collection_exercise_section, "respondents_section": respondents_section}
+
+
+def _build_collection_exercise_section(
+    collection_exercises: list, sample_unit_ref: str, survey_short_name: str, reporting_unit_permission: bool
+) -> list:
+    table = [
+        {
+            "status_class": (
+                CE_STATUS_CLASS[ce["responseStatus"]]
+                if ce["responseStatus"] in CE_STATUS_CLASS.keys()
+                else "ons-status--error"
+            ),
+            "hyperlink": url_for(
+                "case_bp.get_response_statuses",
+                ru_ref=sample_unit_ref,
+                survey=survey_short_name,
+                period=ce["exerciseRef"],
+            ),
+            "hyperlink_text": "Change" if reporting_unit_permission else "View",
+            "period": ce["exerciseRef"],
+            "reporting_unit_name": ce["companyName"],
+            "trading_as": ce["tradingAs"],
+            "region": ce["companyRegion"],
+            "response_status": ce["responseStatus"],
+            "status": _build_ce_status(
+                sample_unit_ref,
+                survey_short_name,
+                ce["exerciseRef"],
+                ce["responseStatus"],
+                reporting_unit_permission,
+            ),
+        }
+        for ce in collection_exercises
+    ]
+    return table
+
+
+def _build_ce_status(ru_ref: str, survey: str, period: str, response_status: str, ru_permission: bool) -> dict:
+    ce_status = {
+        "hyperlink": url_for(
+            "case_bp.get_response_statuses",
+            ru_ref=ru_ref,
+            survey=survey,
+            period=period,
+        ),
+        "hyperlink_text": "Change" if ru_permission else "View",
+        "status_class": (
+            CE_STATUS_CLASS[response_status] if response_status in CE_STATUS_CLASS.keys() else "ons-status--error"
+        ),
+        "response_status": response_status,
+    }
+    return ce_status
+
+
+def _build_respondents_section(
+    respondents: list,
+    case_id: str,
+    collection_exercise_id: str,
+    trading_as: str,
+    ru_name: str,
+    sample_unit_ref: str,
+    ru_id: str,
+    survey_ref: str,
+    survey_short_name: str,
+    survey_id: str,
+    unused_iac: str,
+    permissions: dict,
+) -> list:
+    table = []
+    for respondent in respondents:
+        row = {}
+        if permissions["reporting_unit_edit"]:
+            if unused_iac:
```

Comment:
I see it, the logic is a bit unusual as the frontend is also checking off the iac key being passed to it and diverting it [around](https://github.com/ONSdigital/response-operations-ui/blob/804edb9edfcb540f74c37cce6c6bd8f12de098f4/response_operations_ui/templates/reporting-unit-survey.html#L74). We should try not to do that as it kind of invalidates any unit tests in that area as there is the same logic both in the frontend and back.